### DISCUSSION TITLE
Creating state machines with shared context across instances

### DISCUSSION THREAD

**Discussion by DenniLa2** [Lexpeartha/nuxt-xstate#39] [Q&A] [answered]
Here my test machine.
https://stackblitz.com/edit/nuxt-xstate-playground-jtocz6?file=app.vue,components%2FXTest.vue,components%2Ftest.ts
I want the state to be shared between both components. Tell me how to do it

**Comment by Lexpeartha** [comment, marked as answer]
I think what you want to achieve can be done with sharing the same context between state machines. However that would require some modifications:
```ts
interface TestContext {
  state: ExState;
}

export const sharedContext: TestContext = {
  state: ExState.initial,
};
```
And in your component:
```ts
const machine1 = useMachine(testMachine, { context: sharedContext });
// These instances should share same state
const machine2 = useMachine(testMachine, { context: sharedContext });
```

Of course this in itself won't work, since you would have to update state machine to use `assign()` to update the context, and for that I recommend you check [documentation reference](https://xstate.js.org/docs/guides/context.html#quick-reference)

**Reply by DenniLa2** [reply]
@Lexpeartha You suggest copying the current state into the context? It turns out that I will not be able to share the machine without a common context?

**Reply by Lexpeartha** [reply]
I believe so. Since `useMachine` depends on vue lifecycle hooks, if you're feeling crafty, you might try injecting same instance via nuxt plugin, although I still think using context is better idea.
